.azure-pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -0,0 +1,89 @@
+jobs:
+- job: Build
+  pool:
+    vmImage: vs2017-win2016
+  steps:
+  - task: DotNetCoreInstaller@0
+    displayName: 'Use .NET Core sdk $(DotNetCoreVersion)'
```

Comment:
There isn't any need to add extra escaping for the variable references? Or is that only for the template parameters?

Reply:
Not sure what you mean? I know that sometimes you need to surround a value with quotes IF it breaks the YAML parsing, but this usage seems OK to me. Are you referring to vmIMage or the DotNetCoreVersion?

Reply:
I was meaning like `${{ parameters.name }}` I thought all variables needed to be enclosed like that but perhaps only parameters. I need to spend some more time playing with templates so I can better understand how they are evaluated.

Reply:
I think that is only when you want to propogate variables between jobs that didn't exist at the time the YAML file was processed. For example, if I have a task that produces a variable (calls ##vso[setvariable..]) then that variable is valid across steps within the same job - but to jump across jobs I need to specify that as an output variable for the job - I think.

Reply:
In short - I think this usage is OK.